src/borg/archiver.py

```diff
@@ -533,6 +533,39 @@ def create_inner(archive, cache, fso):
                 else:
                     status = '-'
                 self.print_file_status(status, path)
+            elif args.paths_from_command or args.paths_from_stdin:
+                paths_sep = '\n'
+                if args.paths_delimiter is not None:
+                    # evaluate escape sequences in the separator, in case the shell doesn't replace eg '\\0'
+                    paths_sep = escape_decode(args.paths_delimiter.encode('utf-8'))[0].decode('utf-8')
+                if args.paths_from_command:
+                    try:
+                        proc = subprocess.Popen(args.paths, stdout=subprocess.PIPE)
+                    except (FileNotFoundError, PermissionError) as e:
+                        self.print_error('Failed to execute command: %s', e)
+                        return self.exit_code
+                    pipe = proc.stdout
+                else: # args.paths_from_stdin == True
+                    pipe = sys.stdin
```

Comment:
guess for normal text, this would be fine.

but if there is \0 in there or byte sequences that are not clean unicode: use the binary stream rather?